### PR TITLE
Add missing case for upper save/restore

### LINKED ISSUE #81087: Assertion failed 'unreached' in 'System.Text.Ascii:ChangeCase' when JitDumping
### Description

There are missing entries for `switch (intrinsic->gtIntrinsicName)` in Compiler::gtDispTree.

### Reproduction Steps

Build the testsuite. Then:
```
export DOTNET_JitDump=*
cd /home/alahay01/dotnet/runtime_andy/artifacts/tests/coreclr/linux.arm64.Checked/JIT/Methodical/Methodical_d1
./Methodical_d1.sh
```

### Expected behavior

Should be able to jitdump all methods.

### Actual behavior

```
Assert failure(PID 1930083 [0x001d7363], Thread: 1930083 [0x1d7363]): Assertion failed 'unreached' in 'System.Text.Ascii:ChangeCase[ushort,ushort,System.Text.Ascii+ToLowerConversion](ulong,ulong,ulong):ulong' during 'LSRA allocate' (IL size 1404; hash 0xc6916bbb; FullOpts)

    File: /home/alahay01/dotnet/runtime_andy/src/coreclr/jit/gentree.cpp Line: 12096
    Image: /home/alahay01/dotnet/runtime_andy/artifacts/tests/coreclr/linux.arm64.Checked/Tests/Core_Root/corerun
```

### Regression?

This used to be fine.

### Known Workarounds

Remove the assert :)

### Configuration

Arm64 Linux

### Other information

_No response_

### PR DESCRIPTION
**Pull request by kunalspathak**
Fixes: #81087

### CODE REVIEW

**Comment by kunalspathak** [comment]
@dotnet/jit-contrib @tannergooding

File: src/coreclr/jit/gentree.cpp [hunk 1]
```diff
@@ -12093,7 +12093,14 @@ void Compiler::gtDispTree(GenTree*     tree,
                 case NI_System_Runtime_CompilerServices_RuntimeHelpers_IsKnownConstant:
                     printf(" isKnownConst");
                     break;
-
+#if defined(FEATURE_SIMD)
+                case NI_SIMD_UpperRestore:
+                    printf(" simd_upperRestr");
```
Comment (tannergooding) [inline src/coreclr/jit/gentree.cpp:12098]: nit: 1 extra character to give the full name:
```suggestion
                    printf(" simdUpperRestore");
```

File: src/coreclr/jit/gentree.cpp [hunk 2]
```diff
@@ -12093,7 +12093,14 @@ void Compiler::gtDispTree(GenTree*     tree,
...
+                    break;
+                case NI_SIMD_UpperSave:
+                    printf(" simd_upperSave");
```
Comment (tannergooding) [inline src/coreclr/jit/gentree.cpp:12101]: nit: We can save a character here and make it a bit more readable
```suggestion
                    printf(" simdUpperSave");
```

File: src/coreclr/jit/gentree.cpp [hunk 3]
```diff
@@ -12093,7 +12093,14 @@ void Compiler::gtDispTree(GenTree*     tree,
...
+                    break;
+#endif // FEATURE_SIMD
                 default:
                     unreached();
```
Comment (tannergooding) [inline src/coreclr/jit/gentree.cpp:12105]: Given this is debug only and just impacts JITDUMP, can we do something like `assert` or dump the "raw id" as a fallback?

That is, we could just `printf(" unknown: %d", static_cast<uint16_t>(id))` or something, right?

Comment (tannergooding) [reply src/coreclr/jit/gentree.cpp:12105]: Doing so at least gives usable information and doesn't make getting dumps impossible

**Comment by kunalspathak** [comment]
@tannergooding  - can you review the updated changes?

**Review by tannergooding — APPROVED** [review]

**Pull request merged**